c/cert/src/rules/ARR37-C/DoNotUsePointerArithmeticOnNonArrayObjectPointers.ql

```diff
@@ -13,7 +13,7 @@
 
 import cpp
 import codingstandards.c.cert
-import codingstandards.cpp.dataflow.DataFlow
+import semmle.code.cpp.dataflow.DataFlow
```

Comment:
why are we changing the way that we are currently doing this?
ie previously we were [coping in the lib](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/dataflow/DataFlow2.qll#L11) and avoiding deprecation warnings instead of just accepting them into the tests?

Reply:
The rationale for that change is documented here:
https://github.com/github/codeql-coding-standards/pull/553

TL;DR:
 * The underlying data flow library has changed in a way that breaks our copy (as we only copied a few of the main import files).
 * The reason we retained a copy in https://github.com/github/codeql-coding-standards/commit/6aee03ef5ac577d77fbf034a46a6637f31aefbee was to avoid the absolute paths in the deprecation warnings, but these have now been addressed, so including the deprecation warnings is no longer a problem.

Apologies, I should probably have linked to it (along with other PRs to `next` which are incorporated into this PR).

Reply:
ah ok that makes sense! thanks for the explanation